src/Arcus.Security.Providers.AzureKeyVault/KeyVaultSecretProvider.cs

```diff
@@ -167,6 +175,31 @@ public virtual async Task<Secret> GetSecretAsync(string secretName)
             Guard.NotNullOrWhitespace(secretName, nameof(secretName), "Requires a non-blank secret name to request a secret in Azure Key Vault");
             Guard.For<FormatException>(() => !SecretNameRegex.IsMatch(secretName), "Requires a secret name in the correct format to request a secret in Azure Key Vault, see https://docs.microsoft.com/en-us/azure/key-vault/general/about-keys-secrets-certificates#objects-identifiers-and-versioning");
 
+            if (!_options.TrackDependency)
+            {
+                Secret secret = await GetSecretCoreAsync(secretName);
+                return secret;
+            }
+
+            using (DependencyMeasurement measurement = DependencyMeasurement.Start())
+            {
+                try
+                {
+                    Secret secret = await GetSecretCoreAsync(secretName);
+                    Logger.LogDependency("Azure key vault", secretName, VaultUri, isSuccessful: true, startTime: measurement.StartTime, duration: measurement.Elapsed);
+                    
+                    return secret;
+                }
+                catch
+                {
+                    Logger.LogDependency("Azure key vault", secretName, VaultUri, isSuccessful: false, startTime: measurement.StartTime, duration: measurement.Elapsed);
+                    throw;
+                }
+            }
```

Comment:
This should simplify things a bit:
```suggestion
            bool isSuccessful = false;
            using (DependencyMeasurement measurement = DependencyMeasurement.Start())
            {
                try
                {
                    Secret secret = await GetSecretCoreAsync(secretName);
                    return secret;
                }
                finally
                {
	                if (_options.TrackDependency)
	                {
	                    Logger.LogDependency("Azure key vault", VaultUri, secretName, isSuccessful: isSuccessful, startTime: measurement.StartTime, duration: measurement.Elapsed);
                    }
                }
            }
```

I've swapped the vault & secret name as well, think this will be better for the rendered result.

Can you verify what it looks like please?

Reply:
But we should also track the dependency when the secret retrieval was successful, of course. Otherwise, yeah. You're now also have additional `using` and `try/finally` logic, even when we don't need it, (that's what I was trying to avoid in the original one). But I guess this is also good.

Reply:
It's not that the try/finally and using will make a difference :D

> But we should also track the dependency when the secret retrieval was successful, of course.

This is also included at the top sample, no?

Reply:
My sample was not up-to-date, sorry; changed it to this:
![image](https://user-images.githubusercontent.com/4345663/99035934-f3b99700-2580-11eb-874a-74931b62bfc3.png)

Reply:
Yep, is changed. Thanks.

Reply:
Hmm, question, though, should we include the exception in the context of the failed depenency tracking?

Reply:
Do you mean as dependency data or what do you mean? Do we have an overload for it?

Reply:
Yeah, as dependency data or an entry in our telemetry context (`Dictionary<string, object>`). We don't have an overload for failures explicitly, no.